kpatch-build/lookup.c

```diff
@@ -305,11 +305,11 @@ static void symvers_read(struct lookup_table *table, char *path)
 		if (!symname)
 			perror("strdup");
 
-		objname = strdup(mod);
+		/* Modifies objname in-place */
+		objname = make_modname(mod);
```

Comment:
I don't think this is the right way to fix the issue you are seeing. make_modname() modifies the variable in-place so this way it will modify "mod" and we don't want that.

The pointer returned by basename() in make_modname() is not supposed to be freed, here's an excerpt from the man page:
```
RETURN VALUE
       Both dirname() and basename() return pointers to null-terminated strings.  (Do not pass these pointers to free(3).)
```

Looks like make_modname() needs to be a bit smarter and do something like:
```
tmp = strdup(modname);
<fixup tmp>
ret = strdup(basename(tmp));
free(tmp)
return ret;
```

Reply:
yet, I have changed, please reviev again, thanks!